src/workbookManager.ts

```diff
@@ -230,17 +238,34 @@ export class WorkbookManager {
             }
         });
         (await Promise.all(pivotCachePromises)).forEach(({ path, pivotCacheXmlString }) => {
-            const pivotCacheDoc: Document = parser.parseFromString(pivotCacheXmlString, "text/xml");
-            const element = pivotCacheDoc.getElementsByTagName("cacheSource")[0];
-            if (element.getAttribute("connectionId") == connectionId) {
-                element.parentElement!.setAttribute("refreshOnLoad", refreshOnLoadValue);
-                const newPC = serializer.serializeToString(pivotCacheDoc);
-                zip.file(pivotCachesPath + path, newPC);
+            const {isTableUpdated, newTable} = this.updateTable(pivotCacheXmlString, connectionId, refreshOnOpen, "cacheSource");
```

Comment:
While both are similar, they don't go over the same thing, I would keep updateQueryTable and updatePivotTable separate.

You can see that the actual logic defers, and shared code is mainly the iteration, likely in the future there will be more divergence, so while tempting to merge this query in favor of reuse, its just "by chance" that its reusable.